src/Components/src/Microsoft.AspNetCore.Components/Rendering/ComponentState.cs

```diff
@@ -82,10 +82,16 @@ public void DisposeInBatch(RenderBatchBuilder batchBuilder)
         {
             _componentWasDisposed = true;
  
-            // TODO: Handle components throwing during dispose. Shouldn't break the whole render batch.
             if (_component is IDisposable disposable)
             {
-                disposable.Dispose();
+                try
+                {
+                    disposable.Dispose();
+                }
+                catch
+                {
+                    // Ignore exceptions thrown by components when they're being disposed.
```

Comment:
We should log them somehow though, shouldn't we? If someone's disposal logic is failing they will need a way to know about it. This isn't the entire app shutting down, it's just a user closing a browser tab.

Instead of swallowing them, could we capture them all in an `AggregateException` and then throw it the the end of this `Dispose` method?

Reply:
Actually, could you clarify why this code needs to be changed? It doesn't look like it's involved in the circuit teardown process anyway.

Reply:
I went by the comment that suggested something need to be done about exceptions. We can handle this separately